core/src/main/java/hudson/util/FormValidation.java

```diff
@@ -499,9 +552,9 @@ protected FormValidation handleIOException(String url, IOException e) throws IOE
             // any invalid URL comes here
             if (e.getMessage().equals(url))
                 // Sun JRE (and probably others too) often return just the URL in the error.
-                return error("Unable to connect " + url);
+                return error("Unable to connect " + url, e);
```

Comment:
Minor unrelated change while I was here: append the stack trace for debuggability. We love stack traces.